### PR TITLE
[NFC][SYCL] Minor refactoring in `vector_preview.hpp`

### PR DESCRIPTION
**Pull request by aelovikov-intel**

### CODE REVIEW

File: sycl/include/sycl/vector_preview.hpp [hunk 1]
```diff
@@ -411,8 +409,27 @@ class vec : public detail::vec_arith<DataT, NumElements> {
       T, /*->*/ T>::type;
   // clang-format on
 
-  // convertImpl can't be called with the same From and To types and therefore
-  // we need this version of convert which is mostly no-op.
+  // getValue should be able to operate on different underlying
+  // types: enum cl_float#N , builtin vector float#N, builtin type float.
+  constexpr auto getValue(int Index) const {
```
Comment (aelovikov-intel) [inline sycl/include/sycl/vector_preview.hpp:414]: I'm simply moving it closer to its uses. However, I think these `convert`-related details should be all moved into `vector_convert.hpp` in a later PR.

File: sycl/include/sycl/vector_preview.hpp [hunk 2]
```diff
@@ -148,18 +129,31 @@ class vec : public detail::vec_arith<DataT, NumElements> {
...
 
-public:
 #ifdef __SYCL_DEVICE_ONLY__
+  // clang-format: off
+  using element_type_for_vector_t = typename detail::map_type<
+      DataT,
+#if (!defined(_HAS_STD_BYTE) || _HAS_STD_BYTE != 0)
+      std::byte, /*->*/ std::uint8_t,
+#endif
+      bool, /*->*/ std::int8_t, sycl::half,
```
Comment (uditagarwal97) [inline sycl/include/sycl/vector_preview.hpp:139]: Why the change in formatting? I liked the previous formatting better:
```
bool, /*->*/ std::int8_t,
sycl::half, /*->*/ sycl::detail::half_impl::StorageT,
sycl::ext::oneapi::bfloat16, /*->*/ sycl::ext::oneapi::detail::Bfloat16StorageT,
```

Comment (aelovikov-intel) [reply sycl/include/sycl/vector_preview.hpp]: Wasn't intentional, uploaded new version.

File: sycl/include/sycl/vector_preview.hpp [hunk 3]
```diff
@@ -107,25 +107,6 @@ template <typename T> class GetOp {
   }
 };
 
-// Forward declarations
-template <typename TransformedArgType, int Dims, typename KernelType>
```
[the remark this answers is not in the document]
Comment (aelovikov-intel) [reply sycl/include/sycl/vector_preview.hpp:111]: Wrong question :) You comment implies that these shouldn't be here because you tried to remove them as well. As to some issues - I'm not aware of any but waiting for the CI to finish.

[the remark this answers is not in the document]
Comment (uditagarwal97) [reply sycl/include/sycl/vector_preview.hpp:111]: Yes, they shouldn't be here as they are aren't strictly related to `sycl::vec`.
Here's my theory:
I guess when I splitted `types.hpp` to `vector.hpp` and `vector_preview.hpp`, I incorrectly included these `RoundedRangeKernel` declarations in the vector headers (even `vector.hpp` has these: https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/vector.hpp#L185), thus creating a dependency between vector headers and device copyable code. That's why I was getting an error when trying to remove these from `vector_preview`.

But, when you created the `is_device_copyable` header, you added these declarations to it (https://github.com/intel/llvm/pull/13755/files) and removed the dependency. And so, now these declarations can be removed from `vector_preview.hpp`.

**Review by uditagarwal97 — APPROVED** [review]
